recipes/ulfius/all/conanfile.py

```diff
@@ -0,0 +1,219 @@
+                set_target_properties(Jansson::Jansson PROPERTIES INTERFACE_LINK_LIBRARIES "jansson::jansson")
+                set(JANSSON_VERSION_STRING {self.dependencies['jansson'].ref.version})
+            """), append=True)
+
+        if self.options.with_gnutls:
+            # FIXME: make sure gnutls creates GnuTLSCOnfig.cmake + GnuTLS::GnuTLS target + GNUTLS_VERSION_STRING
+            pass
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", os.path.join(self.source_folder), os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+        save(self, os.path.join(self.package_folder, self._variable_file_rel_path),
+            textwrap.dedent(f"""\
+                set(ULFIUS_VERSION_STRING "{self.version}")
+           """))
+
+        # TODO: to remove in conan v2 once cmake_find_package* generators removed
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {} if self.options.shared else {"Ulfius::Ulfius-static": "Ulfius::Ulfius"}
+        )
+
+    def _create_cmake_module_alias_targets(self, module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent(f"""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """)
+        save(self, module_file, content)
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
+
+    @property
+    def _variable_file_rel_path(self):
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-variables.cmake")
+
+    def package_info(self):
+        libname = "ulfius"
+        if is_msvc(self) and not self.options.shared:
+            libname += "-static"
+        self.cpp_info.libs = [libname]
+
+        target_name = "Ulfius::Ulfius" if self.options.shared else "Ulfius::Ulfius-static"
+        self.cpp_info.set_property("cmake_file_name", "Ulfius")
+        self.cpp_info.set_property("cmake_target_name", target_name)
+        self.cpp_info.set_property("cmake_module_file_name", "Ulfius")
+        self.cpp_info.set_property("cmake_module_target_name", target_name)
```

Comment:
```suggestion
```

conan won't generate a FindUlfius.cmake except if cmake_find_mode is both, but since upstream doesn't provide a FindUlfius.cmake, there is no reason to generate this file.